docs/spark/tutorials/batch-processing.md

```diff
@@ -0,0 +1,150 @@
+
+   ```csharp
+   DataFrame projectsDf = spark
+       .Read()
+       .Schema("id INT, url STRING, owner_id INT, " +
+       "name STRING, descriptor STRING, language STRING, " +
+       "created_at STRING, forked_from INT, deleted STRING" +
+       "updated_at STRING")
+       .Csv("filepath");
+
+   projectsDf.Show();
+   ```
+
+1. Use <xref:Microsoft.Spark.Sql.DataFrame.Na%2A> to drop rows with NA (null) values, and the <xref:Microsoft.Spark.Sql.DataFrame.Drop%2A> method to remove certain columns from your data. This helps prevent errors if you try to analyze null data or columns that are not relevant to your final analysis.
+
+   ```csharp
+   // Drop any rows with NA values
+   DataFrameNaFunctions dropEmptyProjects = projectsDf.Na();
+   DataFrame cleanedProjects = dropEmptyProjects.Drop("any");
+
+   // Remove unnecessary columns
+   cleanedProjects = cleanedProjects.Drop("id", "url", "owner_id");
+   cleanedProjects.Show();
+   ```
+
+## Analyze the data
+
+1. Spark SQL allows you to make SQL calls on your data. It's common to combine user-defined functions and Spark SQL to apply a user-defined function to all rows of your DataFrame.
+
+   You can specifically call `spark.Sql` to mimic standard SQL calls seen in other types of apps. You can also call methods like <xref:Microsoft.Spark.Sql.DataFrame.GroupBy%2A> and <xref:Microsoft.Spark.Sql.DataFrame.Agg%2A> to specifically combine, filter, and perform calculations on your data.
+
+   ```csharp
+   // Average number of times each language has been forked
+   DataFrame groupedDF = cleanedProjects
+       .GroupBy("language")
+       .Agg(Avg(cleanedProjects["forked_from"]) 
+
+   // Sort by most forked languages first
+   groupedDF.OrderBy(Desc("avg(forked_from)")).Show(); 
+
+   spark.Udf().Register<string, bool>(
+       "MyUDF",
+       (date) => DateTime.TryParse(date, out DateTime convertedDate) &&
+           (convertedDate > s_referenceDate);   
+   cleanedProjects.CreateOrReplaceTempView("dateView"); 
+
+   DataFrame dateDf = spark.Sql(
+       "SELECT *, MyUDF(dateView.updated_at) AS datebefore FROM dateView");
+   dateDf.Show();
+   ```
+
+1. Call `spark.Stop()` to end the SparkSession.
+
+## Use spark-submit to run your app
+
+1. Use the following command to build your .NET app:
+
+   ```console
+   dotnet build
+   ```
+
+1. Run your app with spark-submit. Be sure to update the following command with the actual paths to your Microsoft Spark jar file.
+
+   ```console
+   spark-submit --class org.apache.spark.deploy.dotnet.DotnetRunner --master local /<path>/to/microsoft-spark-<version>.jar dotnet /<path>/to/netcoreapp<version>/GitHubProjects.dll
+   ```
+
+## Get the code
+
+You can see the [full solution](https://github.com/dotnet/spark/blob/master/examples/Microsoft.Spark.CSharp.Examples/Sql/Batch/GitHubProjects.cs) on GitHub.
+
+## Additional resources
+
+Provide links to where you can learn more about the topics covered in this tutorial
```

Comment:
Want to remove this section? Or add something to it?

Reply:
I meant to remove it. Thanks!